.dotnet/Azure.AI.OpenAI.Staging/Custom/Audio/AzureAudioClient.Protocol.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using OpenAI;
+using OpenAI.Audio;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
+using System.Text;
+
+namespace Azure.AI.OpenAI.Staging.Audio;
+
+public partial class AzureAudioClient : AudioClient
```

Comment:
I thought these would be internal. Only the top level client would be public.

Reply:
I'll remove these public ones. Instead, I'll update the key-based top-level client constructor to accept a URI:
```
public AzureOpenAIClient(Uri endpoint = null, ApiKeyCredential credential = null, AzureOpenAIClientOptions options = null)
```

The main thing I want to avoid is having the only option for programmatically providing an Azure resource endpoint be using the client options type; that makes sense for the unbranded client, but not for Azure. The above seems to work for that, though, as I can now do this:
```
AzureOpenAIClient topLevelClient = new(new Uri(endpointFromEnvironment), new ApiKeyCredential(keyFromEnvironment));
ChatClient chatClient = topLevelClient.GetChatClient("gpt-35-turbo");
ChatCompletion chatCompletion = chatClient.CompleteChat("hello, world!");
```